sql/plan/create_index.go

```diff
@@ -275,7 +282,55 @@ func (i *loggingKeyValueIter) Next() ([]interface{}, []byte, error) {
 		i.log.Debugf("still creating index: %d rows saved so far", i.rows)
 	}
 
-	return i.iter.Next()
+	_, loc, err := i.iter.Next()
```

Comment:
this feels like too much of a task for `loggingKeyValueIter`, which is just a wrapper for logging

Reply:
Yeah, I also had this feeling (especially that I passed a context, what's not good). But this works. Now we need to refine it, so I'm totally open for any suggestion how to make it internally correctly. Maybe now with indexes we should combine these iterators and use one, more generic. For instance sth. what let me `Eval` expressions without `RowIter`.

Reply:
I think probably the best solution would be to change the API of `IndexKeyValueIter` and change `colNames []string` to `expressions []sql.Expression`. Is there any reason we might not wanna do that? @ajnavarro

Reply:
colNames should be a list of columns because the table doesn't know about expressions, only columns. So, per example, if we want to index `LANGUAGE(f.file_path,f.blob_content)` the IndexKeyValueIterator should return `file_path` and `blob_content` to be able to execute the `LANGUAGE` Function and index it.

Reply:
And this is what it does right now, but index expects to get _languages_, so somewhere we need an `sql.Expression` and call `Eval` but for this we need a `Row`. At the end we also have to pass a `location`. So this is how it looks so far when I added tons of "glue" to workaround and fix it.
The open question is - how to reshape it internally to get rid of this ugly fix.